src/content/platform-integration/ios/app-extensions.md

```diff
@@ -3,62 +3,157 @@ title: Adding iOS app extensions
-   Flutter project directory.
-1. In Xcode, select **File -> New -> Target** from the menu bar. 
+1.  In the console, open the Xcode workspace for the
+    `example_app_with_extension` project.
 
-    <figure>
-      <div class="site-figure-container">
-        <img src='/assets/images/docs/development/platform-integration/app-extensions/xcode-new-target.png' alt='Opening the File -> New menu, then selecting Target in Xcode.' height='300'>
-      </div>
-    </figure>
-1. Select the app extension you intend to add.
-   This selection generates extension-specific code 
-   within a new folder in your project.
-   To learn more about the generated code and the SDKs for each
-   extension point, check out the resources in
-   [Apple's documentation][].
+    ```console
+    $ cd example_app_with_extension && open ios/Runner.xcworkspace
+    ```
 
-To learn how to add a home screen widget to your iOS device,
-check out the 
-[Adding a Home Screen Widget to your Flutter app][lab]
-codelab.
+1.  In Xcode, add an app extension called `Share`
+    and call it `ShareExtension`.
+
+    *   In the Xcode menu bar, select
+        **File** > **New** > **Target**.
+    
+    *   Add **Share Extension**.
+
+    *   In the **Name field**, enter **ShareExtension**.
+
+    *   Click **Finish**.
+
+    *   In the **Activate … Scheme** dialog box that
+        appears, select **Activate**.
+
+1.  In Xcode, change the order of the build process.
+
+    *   Open the **project navigator**
+        (**View** > **Navigators** > **Project**).
+    
+    *   In the **project navigator**, at the top, select
+        **Runner**.
+
+    *   In the main window under **TARGETS**, select
+        **Runner**.
+
+    *   Open the **Build Phases** tab.
+
+    *   Drag **Embed Foundation Extensions** above
+        **Run Script**.
+
+1.  In the console, run the following command to rebuild your
+    iOS app:
+
+    ```console
+    $ flutter build ios --config-only
+    ```
+
+1.  [Test your app with the simulator][].
+
+When you add a new app extension, Xcode generates
+sample code based on the template you selected. For more
+information about the generated code and WidgetKit, see
+[Apple's app extension documentation][].
+
+[Apple's app extension documentation]: {{site.apple-dev}}/app-extensions/
+[Test your app with the simulator]: #test-extensions
+[Share]: https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/Share.html
+
+## Test an iOS app extension {: #test-extensions }
+
+After you've added an app extension to your Flutter project,
+you can test it, using a simulator or physical device.
+If you are testing you extension in debug mode, you must
+use the iOS simulator.
+
+The following steps assume you're using the sample
+application and Share extension from
+[Adding iOS app extensions][].
+
+{% tabs "register-plugins-tabs", true %}
+
+{% tab "Simulator" %}
+
+1.  Add an app extension to your project.
+
+1.  In Xcode, run your iOS app.
```

Comment:
You can alternatively do `flutter run`, which is probably preferred.

Reply:
I like it! Fewer steps. Updated.